keiko-sql/src/main/kotlin/com/netflix/spinnaker/config/SqlQueueConfiguration.kt

```diff
@@ -0,0 +1,64 @@
+package com.netflix.spinnaker.config
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.netflix.spinnaker.q.metrics.EventPublisher
+import com.netflix.spinnaker.q.migration.SerializationMigrator
+import com.netflix.spinnaker.q.sql.SqlDeadMessageHandler
+import com.netflix.spinnaker.q.sql.SqlQueue
+import org.jooq.DSLContext
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty
+import org.springframework.boot.context.properties.EnableConfigurationProperties
+import org.springframework.context.annotation.Bean
+import org.springframework.context.annotation.Configuration
+import java.time.Clock
+import java.time.Duration
+import java.util.Optional
+
+@Configuration
+@EnableConfigurationProperties(SqlQueueProperties::class)
+@ConditionalOnProperty(value = ["keiko.queue.sql.enabled"])
+class SqlQueueConfiguration {
+
+  companion object {
+    const val SCHEMA_VERSION = 1
+  }
+
+  @Bean
+  @ConditionalOnMissingBean(name = ["queue"])
+  fun queue(
+    jooq: DSLContext,
+    clock: Clock,
+    mapper: ObjectMapper,
+    publisher: EventPublisher,
+    serializationMigrator: Optional<SerializationMigrator>,
+    properties: SqlQueueProperties
+  ) =
+    SqlQueue(
+      queueName = properties.queueName,
+      schemaVersion = SCHEMA_VERSION,
+      jooq = jooq,
+      clock = clock,
+      lockTtlSeconds = properties.lockTtlSeconds,
+      mapper = mapper,
+      serializationMigrator = serializationMigrator,
+      ackTimeout = Duration.ofSeconds(properties.ackTimeoutSeconds),
+      deadMessageHandlers = emptyList(), // TODO: implement
```

Comment:
Can we just autowire `Optional<DeadMessageHandler>` or whatever the type is here, and if empty, provide `emptyList`?

Reply:
... Especially since right below we have a dead message handler. 😬 Unless these are different an unfortunately named?

Reply:
Yeah, I had intended to clean this up before opening the PR.